frame/authorship/src/lib.rs

```diff
@@ -314,14 +315,14 @@ impl<T: Trait> Module<T> {
 
 impl<T: Trait> ProvideInherent for Module<T> {
 	type Call = Call<T>;
-	type Error = MakeFatalError<()>;
+	type Error = InherentError;
 	const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
 
 	fn create_inherent(data: &InherentData) -> Option<Self::Call> {
 		let uncles = data.uncles().unwrap_or_default();
 		let mut set_uncles = Vec::new();
 
-		if !uncles.is_empty() {
+		if !uncles.is_empty() && uncles.len() <= MAX_UNCLES {
```

Comment:
Block builder might actually push more uncles than can be included since it can't pre-verify them. This function should just take MAX_UNCLES valid uncles.

Reply:
Right, I guess it's safe here.